presto-main/src/test/java/io/prestosql/block/TestDictionaryBlock.java

```diff
@@ -156,6 +156,23 @@ public void testCompact()
         assertEquals(reCompactedBlock.getDictionarySourceId(), compactBlock.getDictionarySourceId());
     }
 
+    @Test
+    public void testNestedCompact()
+    {
+        Slice[] expectedValues = createExpectedValues(10);
+        DictionaryBlock nestedDictionary = new DictionaryBlock(createSlicesBlock(expectedValues), new int[] {0, 1, 2, 2, 4, 5});
+        DictionaryBlock dictionary = new DictionaryBlock(nestedDictionary, new int[] {2, 3, 2, 0});
+
+        assertEquals(
+                dictionary.getSizeInBytes(),
+                nestedDictionary.getPositionsSizeInBytes(new boolean[] {true, false, true, true, false, false}) + 4 * Integer.BYTES);
+        assertFalse(dictionary.isCompact());
+
+        DictionaryBlock compactBlock = dictionary.compact();
+        assertBlock(compactBlock.getDictionary(), TestDictionaryBlock::createBlockBuilder, new Slice[] {expectedValues[2], expectedValues[0]});
```

Comment:
maybe add `assertInstanceOf(compactBlock.getDictionary(), VariableWidthBlock.class)` to make is super clear to future readers.